W-17738281-App-restart-schedules (#272)
mulesoft/docs-hosting: cloudhub-2/modules/ROOT/pages/ch2-manage-schedules.adoc

When designing the schedule, consider the following:

> * Plan for high availability during network failures. CloudHub 2.0 infrastructure restarts due to hardware issues or scheduled maintenance, and security updates.

Reviewer: ```suggestion
* Plan for high availability during network failures. The CloudHub 2.0 infrastructure restarts due to hardware issues or scheduled maintenance, and security updates.
```

+

> During service updates or security patching, xref:mule-runtime::global-settings-configuration.adoc#global-configurations-reference[CloudHub 2.0 waits five minutes] for existing schedules to complete. After a new replica is launched, the schedule runs at its next due. 

Reviewer: ```suggestion
During service updates or security patching, xref:mule-runtime::global-settings-configuration.adoc#global-configurations-reference[CloudHub 2.0 waits five minutes] for existing schedules to complete. After a new replica is launched, the schedule runs at its next scheduled time. 
```

Reviewer: Why the inline link? I think in this case, it would be better to have a "For more information about <whatever this link is meant to convey>, see xref:mule-runtime::global-settings-configuration.adoc#global-configurations-reference[Global Configurations Reference].

Reviewer 2: this should be the same behavior as described in cloudhub-2/modules/ROOT/pages/ch2-understand-app-restart.adoc. - Cloudhub 2.0 waits at max five minutes only if shutdownTimeout is configured properly

> * Design your schedules to prevent duplicate data processing, as they can be triggered more than once due to the behavior mentioned previously.

Reviewer: Specifically which previously mentioned behavior? 

Should this be something like:
* Because schedules can be triggered more than once, design them to prevent duplicate data processing.

For example, what would they do to prevent this?

> If clustering is enabled, CloudHub 2.0 ensures the schedule is executed on a single replica, regardless of how many replicas are running. Subsequent schedule executions can occur on a different replica. CloudHub 2.0 distributes multiple concurrent schedules across replicas.

Reviewer: ```suggestion
If clustering is enabled, CloudHub 2.0 executes the schedule on a single replica, regardless of how many replicas are running. Subsequent schedule executions can occur on a different replica. CloudHub 2.0 distributes multiple concurrent schedules across replicas.
```

* If clustering isn't enabled, schedules run individually on all replicas.

> * For a fixed frequency, the scheduler’s xref:cloudhub-2::ch2-manage-schedules.adoc#disable-enable-scheduler[minimum recommended start delay is five seconds].

Reviewer: ```suggestion
* For a fixed-frequency scheduler, the xref:cloudhub-2::ch2-manage-schedules.adoc#disable-enable-scheduler[minimum recommended start delay is five seconds].
```

Reviewer: Again, it's not clear what the inline link is providing - it would be better to give context and say "For more information about...".

> * If a schedule isn't executed because the application isn't running, CloudHub 2.0 doesn't trigger it immediately when the application starts. The schedule is executed at its next due date.

Reviewer: due date = scheduled time?

* For multi-replica apps, CloudHub 2.0 runs scheduled jobs on all replicas.
+
If you run the scheduled job immediately by clicking *Run*, the scheduler runs on a single replica.
+
CloudHub 2.0 distributes multiple concurrent schedulers across replicas.
* You can't manage CloudHub 2.0 schedules using scripts.
+
Use the *Schedules* tab or the CloudHub 2.0 API to manage CloudHub 2.0 schedules.
* Flow names can contain only upper and lower case letters, numbers, hyphens (`-`), underscores (`_`), and periods (`.`). The following characters are invalid: `/`, `[`, `]`, `{`, `}`, and `#`.
* When you create a schedule using a scheduler that has a placeholder, the value that appears when you create a schedule varies based on when the placeholder is discovered. For example:
** If the value is available before the app is deployed, the value appears.
** If the value is available before deployment and depends on deployment properties, an estimation of the value appears.
** If the value is a placeholder (such as `${cronExpression}`) that is provided at the time of deployment, the exact value of the placeholder doesn't appear in the user interface.
** CloudHub 2.0 uses a default value of `60000`.

> * If you change schedulers, reupload the application to Exchange to make sure the expected scheduler behavior occurs.

Reviewer: ```suggestion
* If you change a scheduler, reupload the application to Exchange to make sure the expected scheduler behavior occurs.
```
